openedx/core/djangoapps/xblock/tests/test_utils.py

```diff
@@ -0,0 +1,108 @@
+"""
+Tests for xblock utils.
+"""
+import datetime
+
+import pytest
+from django.test import override_settings
+from freezegun import freeze_time
+
+from openedx.core.djangoapps.xblock.utils import (
+    get_secure_token_for_xblock_handler,
+    validate_secure_token_for_xblock_handler
+)
+
+REFERENCE_PARAMS = {
+    "generation_user_id": "12344",
+    "validation_user_id": "12344",
+    "generation_block_key_str": "some key",
+    "validation_block_key_str": "some key",
+    "generation_secret_key": "baseline_key",
+    "validation_secret_key": "baseline_key",
+    "generation_xblock_handler_token_keys": None,
+    "validation_xblock_handler_token_keys": None,
+    "reference_time": datetime.datetime(2021, 1, 28, 13, 26, 38, 787309),
+    "validation_time_delta_s": 0,
+}
+
+
+@pytest.mark.parametrize(
+    "param_delta,expected_validation",
+    [
+        ({}, True),
+        # Ensure token still valid in 1 hour
+        ({"validation_time_delta_s": 3600}, True),
+        # Ensure token still valid in 1 day, this should be true but isn't due
+        # to the bug that will be fixed in ARCHBOM-1677
+        ({"validation_time_delta_s": 86400}, False),
+        # Ensure token is invalid after 5 days
+        ({"validation_time_delta_s": 86400 * 5}, False),
+        # Ensure token is not valid 5 days before generation
+        # In the case where the validating server is really skewed
+        # from the generating server.
+        ({"validation_time_delta_s": 86400 * -5}, False),
+        # Different user tries to use your token.
+        ({"validation_user_id": 54321}, False),
+        # Access a different block.
+        ({"validation_block_key_str": "some other block"}, False),
+        # If key is changed, token shouldn't be valid.
+        ({"validation_secret_key": "new secret key"}, False),
+        # Test with token keys setting overrides that don't have anything to do with the secret key.
+        ({"generation_xblock_handler_token_keys": [], "validation_xblock_handler_token_keys": []}, True),
+        # Test migration from secret key to keys list. The secret_key has changed
+        # but the old secret key is in the token keys list.
+        (
+            {
+                "validation_xblock_handler_token_keys": ["baseline_key", ],
+                "validation_secret_key": "new secret key",
+            },
+            True,
+        ),
+        # Test tokens generated with the old key will be valid when there is a new primary
+        # token key at validation time.
+        (
+            {
+                "generation_xblock_handler_token_keys": ["baseline_key", ],
+                "validation_xblock_handler_token_keys": ["now token key", "baseline_key", ],
```

Comment:
```suggestion
                "validation_xblock_handler_token_keys": ["new token key", "baseline_key", ],
```
Typo, although shouldn't change outcome.